src/nix/store-check-trusted.cc

```diff
@@ -0,0 +1,61 @@
+#include <sstream>
+
+#include "command.hh"
+#include "logging.hh"
+#include "serve-protocol.hh"
+#include "shared.hh"
+#include "store-api.hh"
+#include "local-fs-store.hh"
+#include "util.hh"
+#include "worker-protocol.hh"
+
+using namespace nix;
+
+namespace {
+
+bool checkPass(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_GREEN "[PASS] " ANSI_NORMAL + msg);
+    return true;
+}
+
+bool checkFail(const std::string & msg) {
+    if (isatty(fileno(stdin)))
+      notice(ANSI_RED "[FAIL] " ANSI_NORMAL + msg);
+    return false;
+}
+
+}
+
+struct CmdCheckTrusted : StoreCommand
+{
+    bool success = true;
+
+    Category category() override { return catUtility; }
+
+    std::string description() override
+    {
+        return "ask a Nix store if you are a trusted-user, return exit code 0 if trusted, 1 if not trusted, 2 if unexpected error";
+    }
```

Comment:
Better keep the description short and add some proper doc of how this works, something like:

```suggestion
    std::string description() override
    {
        return "check whether you are a trusted-user";
    }
    
    std::string doc() override
    {
          return                                                                                                                                     
              #include "store-check-trusted.md"                                                                                                                     
              ;    
    }
```

Where `store-check-trusted.md` would contain some more docs for the command (similar to https://github.com/NixOS/nix/blob/master/src/nix/ping-store.md for example)

Reply:
> proper doc of how this works

The proper doc of how this works is that it returns 1, 0 or 2. I don't think it's possible to make a dedicated document, unless you want it to describe the implementation details, which I think would not be suitable.

Reply:
Are there other commands with similarly very short and possibly useless manual pages? I think this would actually hide the information? `nix doctor` does not have its own man page, for the same reason, it's too small to document.

Reply:
> The proper doc of how this works is that it returns 1, 0 or 2. I don't think it's possible to make a dedicated document, unless you want it to describe the implementation details, which I think would not be suitable.

Fair, a separate document might be overkill. But keeping the description short and moving the exact behaviour separate (even a simple one like here) is important for things like `nix --help` (which will list all the commands with their description).

Reply:
After breaking it out, I found that showing an example with `--store` is beneficial anyway and have created the md file.